setup.py

```diff
@@ -165,7 +165,11 @@ def get_git_sha() -> str:
         "firebolt": ["firebolt-sqlalchemy>=0.0.1"],
         "gsheets": ["shillelagh[gsheetsapi]>=1.2.6, <2"],
         "hana": ["hdbcli==2.4.162", "sqlalchemy_hana==0.4.0"],
-        "hive": ["pyhive[hive]>=0.6.5", "tableschema", "thrift>=0.14.1, <1.0.0"],
+        "hive": [
+            "pyhive[hive_pure_sasl]>=0.7.0",
```

Comment:
@mdeshmu would you mind explaining the difference between the `hive` and `hive_pure_sasl` extra installs?

Reply:
'pyhive[hive]' extras uses [sasl](https://pypi.org/project/sasl/) that doesn't support Python 3.11, See https://github.com/cloudera/python-sasl/issues/30. Hence PyHive also supports [pure-sasl](https://pypi.org/project/pure-sasl/) via additional extras 'pyhive[hive_pure_sasl]' which support Python 3.11.

Reply:
I would suggest to break it into 2 options e.g. smth like:
"hive": ... # py 3.10 only
"hive_pure_sasl": ... any py version

Reply:
@bkyryliuk I have updated the PR.

pure-sasl library is installed in all PY versions since it is a dependency of thrift-sasl which is a dependency of PyHive.

PY < 3.11: sasl library will be installed and used as PyHive's code gives it first preference. 
PY >= 3.11 Only pure-sasl library will be installed and used.

`test-postgres-hive` CI test in this repo will always pure-sasl because PY 11 test cases fail if sasl is installed.

Reply:
s/pyhive[hive]>=0.7.0;python_version<'3.11'/pyhive[hive]>=0.6.5;python_version<'3.11'

- unless there is a good reason to force un update for the superset users
- same for the spark below

Reply:
done